src/plugins/teleport-plugin-react-base-component/utils.ts

```diff
@@ -293,20 +289,29 @@ const makeStateHookAST = (stateIdentifier: StateIdentifier, t = types) => {
 }
 
 const makeRepeatStructureWithMap = (
-  dataSource: string | any[],
+  dataSource: UIDLNodeAttributeValue,
   content: types.JSXElement,
   meta: Record<string, any> = {},
   t = types
 ) => {
   const iteratorName = meta.iteratorName || 'item'
   const keyIdentifier = meta.useIndex ? 'index' : iteratorName
 
-  const source =
-    typeof dataSource === 'string'
-      ? t.identifier(dataSource)
-      : t.arrayExpression(dataSource.map((element) => convertValueToLiteral(element)))
+  let source
+
+  if (dataSource.type === 'static') {
+    source = t.arrayExpression(
+      (dataSource.content as any[]).map((element) => convertValueToLiteral(element))
+    )
+  } else {
```

Comment:
This else implies all types of entities but 'static'. When we add a third one we'll get fuzzy errors here without a default throw error for not recognized type

Reply:
good idea, will change this